client/app/components/category_list.vue

```diff
@@ -1,5 +1,94 @@
 <template lang="pug">
-    main(role='contentinfo')
-        h1 {{ 'my_accounts category title' | t }}
-        article {{ 'my_accounts coming_soon' | t }}
+    extends ../templates/layout.pug
+
+    block content
+        cozy-dialog(v-for="item in dialogs", v-bind:item="item", v-on:success="onSuccess", v-on:error="onError", v-on:submit="onSubmit", v-on:close="onCloseDialog")
+
+        include ../templates/category_list.pug
+
+        ul
+            li(v-for="link in links")
+                button(v-on:click="onOpenDialog()")
+                  | {{ link.message }}
 </template>
+
+<script>
+    const dialogs = [{
+        id: 'dialog-1',
+        hidden: false,
+        headerImage: 'test0.png',
+        success: 'create-account-success',
+        error: 'create-account-error'
+    }]
+
+
+    export default {
+        data () {
+            return {
+                dialogs: [],
+                links: [
+                  { message: 'open dialog' }
+                ]
+            }
+        },
+
+
+        methods: {
+            onOpenDialog (item) {
+                if (item === undefined) {
+                    item = dialogs[0]
+                }
+
+                // Show <dialog>
+                this.dialogs.push(item)
+
+                // Update URL query
+                this.$router.push({ query: { dialog: item.id }})
+            },
+
+
+            onCloseDialog (item) {
+                // Hide <dialog>
+                const index = this.dialogs.indexOf(item)
+                this.dialogs = this.dialogs.splice(index, 0)
+
+                // Update URL query
+                this.$router.push({ query: {}})
```

Comment:
I'm not sure that updating the query should be the responsability of the `onClose` hook. IMO, it's the dialog that updates the query param when invoking its internal `close` method, which occured a rendering of the `app.vue` top component (which listen to query param changes), and then destroy the modal.